src/test/java/seedu/address/model/person/UniquePersonListTest.java

```diff
@@ -55,29 +56,29 @@ public void add_nullPerson_throwsNullPointerException() {
     @Test
     public void add_duplicatePerson_throwsDuplicatePersonException() {
         uniquePersonList.add(ALICE);
-        assertThrows(DuplicatePersonException.class, () -> uniquePersonList.add(ALICE));
+        assertThrows(DuplicateElementException.class, () -> uniquePersonList.add(ALICE));
```

Comment:
Why not just test UniqueElementList rather than testing each implementation of it?

Reply:
because isSameElement changes depending on the class of T